### ISSUE TITLE
NeoVim Cursor dissapears when `NVIM_TUI_ENABLE_TRUE_COLOR=1`

### ISSUE THREAD

**Issue by jimmyharris** [neovim/neovim#3692]
I am using [MobaXterm 8.3](http://mobaxterm.mobatek.net/download-home-edition.html) which is based on modified versions of putty listed in [this gist](https://gist.github.com/XVilka/8346728). By default I always enable true color support using `NVIM_TUI_ENABLE_TRUE_COLOR=1` before launching nvim from the terminal. By default, When MobaXterm 8.3 is set to use the block cursor display the cursor disappears.  The only way to make it visible is to set the cursor to use the "underline" setting. Do we have any idea why NeoVim is not drawing the cursor in this mode? The cursor displays correctly using XTerm and iTerm. It also displays correctly when true color mode is disabled.

Any idea what could be going on here?

**Comment by jimmyharris** [comment]
I have a theory, and that theory mostly has to do with the fact that NeoVim make use of `has('gui_running')` when the TUI is in true-color mode to inform colorschemes that we are using gui colors instead of console colors.  I will see if I can get a work around going.

**Comment by justinmk** [comment]
If theory is correct, latest master should help here. See https://github.com/neovim/neovim/pull/4155

**Comment by metincc** [comment]
I'm using NVIM 0.2.0-dev, mobaXterm 9.4 and 'morhetz/gruvbox'. Cursor is not visible when NVIM_TUI_ENABLE_TRUE_COLOR=1 and I couldn't make it visible trying many things including has('gui_running') and setting NVIM_TUI_ENABLE_CURSOR_SHAPE different values. Is there any solution for this problem? I'm willing to do any work around or hack.

**Comment by justinmk** [comment]
Check the FAQ

**Comment by metincc** [comment]
Thank you for quick response. Actually, I had checked the FAQ. Please let me share some more info. Issue happens when I do ssh to my machine via mobaXterm. When I log on the machine directly and when I use Konsole to start neovim, there is no issue. Also there is no issue when I do ssh with mobaXterm and use Vim 7.4.52 with the same vimrc and I have visible cursor. Cursor disappears only with Neovim, not Vim.

**Comment by justinmk** [comment]
See the note about `NVIM_TUI_ENABLE_CURSOR_SHAPE` in `man nvim`. If `let $NVIM_TUI_ENABLE_CURSOR_SHAPE=0` doesn't help then that's very unusual.

Note that `NVIM_TUI_ENABLE_TRUE_COLOR` is ignored, see https://github.com/neovim/neovim/wiki/Following-HEAD#20160511

**Comment by metincc** [comment]
Right, my vimrc has 'let $NVIM_TUI_ENABLE_CURSOR_SHAPE=0' and 'set termguicolors'. And I've also tried 1 and 2 for 'NVIM_TUI_ENABLE_CURSOR_SHAPE', issue persists.

**Comment by justinmk** [comment]
Don't know. Need more info, or a patch.

**Comment by metincc** [comment]
nvim -u NORC: Cursor is visible but 
![image](https://cloud.githubusercontent.com/assets/11456473/23844429/c6cc1c18-07ca-11e7-971a-a3103be713a0.png)

**Comment by metincc** [comment]
I've switched to cygwin from mobaXterm. It looks like everything works as I expected in cygwin.
![image](https://cloud.githubusercontent.com/assets/11456473/23888568/5a2cc73e-0891-11e7-8202-49cafe42a9b4.png)

**Comment by zodiacg** [comment]
I have exactly the same problem with nvim & mobaxterm. nvim -u NORC works fine (I'm using SpaceVim as my rc).
I tried clean guicursor option but not helping.

**Comment by cazador481** [comment]
I found that I need to remove the following lines out of my init file for my cursor to show up.
```vim
if has("multi_byte")  
         if &termencoding == ""
             let &termencoding = &encoding
         endif
 endif 
```

**Comment by bfredl** [comment]
@cazador481 Are you sure that was the only change you did at a time? Because `&tenc` is currently ignored by nvim.

**Comment by cazador481** [comment]
@bfredl: That was the only change that I did.  Alas, I can't reproduce it now, as I have upgraded my neovim version.

**Comment by ericdorsey** [comment]
> I am using Mobaxterm v10.9. I have also met this problem.

FWIW, I ran across this thread because I was trying to fix the gruvbox Vim colorscheme having muted colors -- using `set termguicolors` and the cursor would disappear, so even though the colors would be vivid after that, Vim was basically unusable due to the cursor missing.

It appears this issue is resolved in MobaXterm v12.1 Build 4156 as I no longer have this problem after upgrading.
